corehq/apps/users/management/commands/add_data_dict_permissions.py

```diff
@@ -0,0 +1,73 @@
+from django.core.management.base import BaseCommand
+
+from django.db.models import Q
+
+from dimagi.utils.chunked import chunked
+from corehq.apps.users.permissions import EXPORT_PERMISSIONS
+from corehq.apps.users.models_role import Permission, UserRole
+from corehq.toggles import DATA_DICTIONARY, DATA_FILE_DOWNLOAD
+
+
+class Command(BaseCommand):
+    help = "Adds data dictionary permission to user role if not already present and edit tab is viewable."
+
+    def build_role_can_edit_commcare_data_q_object(self):
+        edit_data_permission, created = Permission.objects.get_or_create(value='edit_data')
+        return Q(rolepermission__permission_fk_id=edit_data_permission.id)
+
+    def build_role_can_export_data_q_object(self):
```

Comment:
Just to make sure i'm understanding correctly - the original query would incorrectly include roles that have access to only reports that are not in `EXPORT PERMISSIONS`. This is because the "view_reports" permission will be assigned to "role_permission" even if it's not an allow all access.

Reply:
yep, I believe that to be correct, though it couldn't hurt to verify

Reply:
corrected in 6671e78bdaa0d8e9fe385c9e53c09b57b33302a5
The associated test `test_role_has_download_data_files_permission' did fail the previous query but does pass this new query. Thank you for catching this.